src/lib/adapters/waku/index.ts

```diff
@@ -209,40 +238,57 @@ export default class WakuAdapter implements Adapter {
 	async onLogIn(wallet: BaseWallet): Promise<void> {
 		const address = wallet.address
 
+		const ownPrivateKey = wallet.privateKey
+		const ownPublicKey = wallet.signingKey.compressedPublicKey
+
+		const ownPublicEncryptionKey = hexToBytes(hashHex(ownPublicKey))
+		const ownPrivateEncryptionKey = hexToBytes(hashHex(wallet.privateKey))
```

Comment:
This will not work with external wallets like MetaMask. Not necessarily a huge problem, but good to keep in mind in case we want to support these at some point.

Also, what's the point of encrypting the `localStorage` if you can just get the wallet's private key and decrypt it anyways? Or where is the wallet stored?

Reply:
Good point regarding external wallets.

When you disconnect the mnemonic is deleted from the localstorage, but your other data is not. This way if you are paranoid you can use the app by connecting with your mnemonic and disconnecting when you are done, but your data may stay in the localstorage and you can still keep you contacts etc. Or you can have multiple accounts and change between them easily.

The alternative would be to delete everything from the localstorage when you disconnect, but that would make multiple accounts less convenient, so I choose to use encryption instead.